### DISCUSSION TITLE
Wyze Settings Without Wyze app

### DISCUSSION THREAD

**Discussion by jsolarin** [gtxaspec/wz_mini_hacks#309] [Development]
I was able to reverse the /configs/.user_config settings so the Wyze app is not needed if you remove the camera or don't have internet access.

Hopefully someone finds this useful, or if the wz_mini web interface could display this. I don't know of a way to HUP the iCamera process to get it to reload the settings, so a reboot is needed to read the settings atm. (Only touch the SETTING section of the config, the rest has some generated hashes)

```
[SETTING]
fps=30				# Frames Per Second
bitRate=120			# (60=360p or SD, 120=HD)
res=1				# (1=SD or HD, 2=360p)
timezone=-4			# Timezone offset from GMT
drawBoxSwitch=2			# Motion Box (1=Enabled, 2=Disabled)

# Dection Settings
AMALevel=2			# Audio Detection Sensitivity (2-252)
MMALevel=2			# Motion Detection Sensitivity (2-252)
#MAT=0				# Detection Zone (1=Enabled, 0=Disabled)
#AALX=12			# Detection Zone X
#AASX=37			# Detection Zone X
#AALY=44			# Detection Zone Y
#AASY=22			# Detection Zone Y

# Event Recording
MASwitch=2			# Detects Motion (1=Enabled, 2=Disabled)
AASwitch=2			# Detects Sound (1=Enabled, 2=Disabled)

# Alarm Settings
SASwitch=2			# Smoke Alarm Sound (1=Enabled, 2=Disabled)
CASwitch=2			# CO Alarm Sound (1=Enabled, 2=Disabled)

# Advanced Settings
recordType=1			# Record to SD (1=Continous, 2=Events Only, 3=Disabled)

nightVision=3			# Night Vision Mode (1=On, 2=Off, 3=Auto)
night_cut_thr=1			# Nightvision Conditions (1=Dusk, 2=Dark)
night_led_ex=1			# Night Vision IR Lights Near (1=Near, 2=Off or Far)
NIGHT_LED_flag=2		# Night Vision IR Lights Far (1=Far, 2=Off or Near)

indicator=2			# Camera Status Light (1=Enabled, 2=Disabled)
verSwitch=1			# Flip Vertical (1=Disabled, 2=Enabled)
horSwitch=1			# Flip Horizontal (1=Disabled, 2=Enabled)
osdSwitch=2			# Show Timestamp (1=Enabled, 2=Disabled)
watermark_flag=2		# Show WYZE Logo (1=Enabled, 2=Disabled)
AST=2				# Record Sound (1=Enabled, 2=Disabled)
```

**Comment by kohrar** [comment]
Excellent work! 

I've made some notes on some of these options for myself, but I wasn't as diligent as you when figuring out what each option did. 

It'd be interesting to see if the bitrate value can be set to some other values, or is it only 60/120 (SD/HD)? 

There are other options that can be set as well. If you look at the iCamera start messages, you can see these keys and the values that were set. This is what I have on my v3. 

```
[user_config.c:373][USER SET] [indicator    ] [2]
[user_config.c:373][USER SET] [nightVision  ] [3]
[user_config.c:373][USER SET] [bitRate      ] [120]
[user_config.c:373][USER SET] [res          ] [1]
[user_config.c:373][USER SET] [fps          ] [20]
[user_config.c:373][USER SET] [horSwitch    ] [2]
[user_config.c:373][USER SET] [verSwitch    ] [2]
[user_config.c:373][USER SET] [osdSwitch    ] [1]
[user_config.c:373][USER SET] [logSd        ] [1]
[user_config.c:373][USER SET] [logUdisk     ] [1]
[user_config.c:373][USER SET] [telntSwitch  ] [2]
[user_config.c:373][USER SET] [recordType   ] [3]
[user_config.c:373][USER SET] [MASwitch     ] [2]
[user_config.c:373][USER SET] [MALevel      ] [5]
[user_config.c:373][USER SET] [AASwitch     ] [2]
[user_config.c:373][USER SET] [AALevel      ] [5]
[user_config.c:373][USER SET] [SASwitch     ] [2]
[user_config.c:373][USER SET] [CASwitch     ] [2]
[user_config.c:373][USER SET] [TBStart      ] [0]
[user_config.c:373][USER SET] [TBDuration   ] [1440]
[user_config.c:373][USER SET] [drawBoxSwitch] [2]
[user_config.c:360][USER SET] [timezone     ] [-6]
[user_config.c:373][USER SET] [MMALevel     ] [100]
[user_config.c:373][USER SET] [AMALevel     ] [100]
[user_config.c:373][USER SET] [MAT          ] [0]
[user_config.c:373][USER SET] [AST          ] [1]
[user_config.c:373][USER SET] [motionTrack  ] [2]
[user_config.c:373][USER SET] [motionCruisin] [2]
[user_config.c:373][USER SET] [AASX         ] [25]
[user_config.c:373][USER SET] [AASY         ] [25]
[user_config.c:373][USER SET] [AALX         ] [50]
[user_config.c:373][USER SET] [AALY         ] [50]
[user_config.c:373][USER SET] [motionbackvalue] [1]
[user_config.c:373][USER SET] [slide_x      ] [0]
[user_config.c:373][USER SET] [slide_y      ] [0]
[user_config.c:373][USER SET] [preset_x     ] [0]
[user_config.c:373][USER SET] [preset_y     ] [0]
[user_config.c:373][USER SET] [ubootflag    ] [0]
[user_config.c:373][USER SET] [exposureModel] [0]
[user_config.c:373][USER SET] [limit_max_x  ] [0]
[user_config.c:373][USER SET] [limit_min_x  ] [0]
[user_config.c:373][USER SET] [limit_max_y  ] [0]
[user_config.c:373][USER SET] [limit_min_y  ] [0]
[user_config.c:373][USER SET] [motor_limit_flag] [0]
[user_config.c:373][USER SET] [resolution_select_flag] [1]
[user_config.c:373][USER SET] [low_bitrate_num] [30]
[user_config.c:373][USER SET] [watermark_flag] [2]
[user_config.c:373][USER SET] [speaker_vol  ] [50]
[user_config.c:373][USER SET] [pir_alarm_switch] [0]
[user_config.c:373][USER SET] [NIGHT_LED_flag] [2]
[user_config.c:373][USER SET] [sdcard_exist ] [0]
[user_config.c:373][USER SET] [autoRecordNum] [0]
[user_config.c:373][USER SET] [autoRecordMem] [0]
[user_config.c:373][USER SET] [dongle_led   ] [1]
[user_config.c:373][USER SET] [alarmDate    ] [255]
[user_config.c:373][USER SET] [alarmInterval] [300]
[user_config.c:373][USER SET] [SDEVICE_ACTION] [0]
[user_config.c:373][USER SET] [voice_switch ] [2]
[user_config.c:373][USER SET] [voice_scene  ] [1]
[user_config.c:373][USER SET] [auto_clarity ] [2]
[user_config.c:373][USER SET] [cmc_upload   ] [1]
[user_config.c:373][USER SET] [p2p_log      ] [2]
[user_config.c:373][USER SET] [night_led_ex ] [2]
[user_config.c:373][USER SET] [night_cut_thr] [1]
[user_config.c:373][USER SET] [slightEnable ] [1]
[user_config.c:373][USER SET] [slightMode   ] [2]
[user_config.c:373][USER SET] [slightLowLight] [2]
[user_config.c:373][USER SET] [slightBright ] [255]
[user_config.c:373][USER SET] [hms_enable   ] [0]
[user_config.c:373][USER SET] [hms_state    ] [0]
[user_config.c:373][USER SET] [slightPreemptSw] [0]
```

**Reply by mrlt8** [reply]
> It'd be interesting to see if the bitrate value can be set to some other values, or is it only 60/120 (SD/HD)?

Some of the newer cameras default at 180 for "HD", but you can crank it up to 255.

The resolution is a little different depending on the camera:  1 for 1080p and 2 for 360p on "normal" cameras (v1 only supports 1080p), 3 for 1728 and 4 for 640 on doorbells, and I believe 4 for 2K on the "pro" cameras.

I could never figure out the "telntSwitch", but "alarmInterval" should also let you reduce the cooldown time from the standard 5 min.

**Reply by jsolarin** [reply]
Thanks! Any idea or mention in the log on how to reload the config or is it only read on startup?

That would be the last piece to writing this into the wz_mini webserver and ditch the app.

**Reply by mrlt8** [reply]
Not sure. Changes can be pushed over the tutk protocol, but I believe they're only read from the config on startup?

**Reply by virmaior** [reply]
@mrlt8  based on https://github.com/gtxaspec/wz_mini_hacks/blob/1fee46a051429086f6e0b7b577100ac25645c72c/SD_ROOT/wz_mini/bin/cmd , some things seem to be accessible by sending to port 4000 via nc .

**Comment by virmaior** [comment]
these are excellent finds. I've been thinking to ask how to figure exactly these settings because ditching anything wyze is definitely a solid objective. the current config page basically shadows wz_mini.conf so maybe these settings can go on a different page?

When I get a bit of time, I'll try to add this sort of functionality to the web interface.

another piece of the pie I've got cooking is a screen to show multiple cameras at once. Right now, I can use it to see 5 cameras at 1 frame/second jpeg. In my case they are through a rpi http_proxy setup but I'm trying to set it up for other applications as well.

**Comment by kohrar** [comment]
I guess as reported in #38, the v2 stores the configs elsewhere. I didn't notice this until this was brought up.

User configs are set in `/configs/.parameters` on the v2. Here're the keys I see in the v2's iCamera log messages for future reference.


```
<Reading> params from /configs/.parameters
[USERSET] [indicator    ] [1]
[USERSET] [nightVision  ] [3]
[USERSET] [bitRate      ] [100]
[USERSET] [res          ] [1]
[USERSET] [fps          ] [15]
[USERSET] [horSwitch    ] [1]
[USERSET] [verSwitch    ] [1]
[USERSET] [osdSwitch    ] [1]
[USERSET] [logSd        ] [1]
[USERSET] [logUdisk     ] [1]
[USERSET] [telntSwitch  ] [2]
[USERSET] [recordType   ] [1]
[USERSET] [MASwitch     ] [2]
[USERSET] [MALevel      ] [5]
[USERSET] [AASwitch     ] [0]
[USERSET] [AALevel      ] [5]
[USERSET] [SASwitch     ] [0]
[USERSET] [CASwitch     ] [0]
[USERSET] [TBStart      ] [0]
[USERSET] [TBDuration   ] [1440]
[USERSET] [drawBoxSwitch] [1]
[USERSET] [timezone     ] [0]
[USERSET] [MMALevel     ] [125]
[USERSET] [AMALevel     ] [125]
[USERSET] [MAT          ] [0]
[USERSET] [AST          ] [1]
[USERSET] [motionTrack  ] [2]
[USERSET] [motionCruisin] [2]
[USERSET] [AASX         ] [25]
[USERSET] [AASY         ] [25]
[USERSET] [AALX         ] [50]
[USERSET] [AALY         ] [50]
[USERSET] [motionbackvalue] [1]
[USERSET] [slide_x      ] [0]
[USERSET] [slide_y      ] [0]
[USERSET] [preset_x     ] [0]
[USERSET] [preset_y     ] [0]
[USERSET] [ubootflag    ] [0]
[USERSET] [exposureModel] [0]
[USERSET] [limit_max_x  ] [0]
[USERSET] [limit_min_x  ] [0]
[USERSET] [limit_max_y  ] [0]
[USERSET] [limit_min_y  ] [0]
[USERSET] [motor_limit_flag] [0]
[USERSET] [resolution_select_flag] [1]
[USERSET] [low_bitrate_num] [30]
[USERSET] [watermark_flag] [1]
[USERSET] [speaker_vol  ] [50]
[USERSET] [pir_alarm_switch] [0]
[USERSET] [NIGHT_LED_flag] [1]
[USERSET] [sdcard_exist ] [0]
[USERSET] [autoRecordNum] [0]
[USERSET] [autoRecordMem] [0]
[USERSET] [alarmDate    ] [255]
[USERSET] [SDEVICE_ACTION] [0]
[USERSET] [hms_enable   ] [0]
[USERSET] [hms_state    ] [0]
[USERSET] [CamPlus      ] [0]
[USERSET] [EdgeAI       ] [0]
[USERSET] [ObjectDetection] [0]
[USERSET] [FaceRecognition] [0]
[USERSET] [EventInsight ] [0]
[USERSET] [SoundDetection] [0]
```

**Comment by virmaior** [comment]
I've put in a pull request that adds /cgi-bin/cam.cgi
which at least can display the info and some of the tips.
I'm a bit wary to mess with the settings myself (since I don't have time to fix it if I break it right now).

Also two very major limitations:
1. I'm still running an old version of wyze firmware on my v3s and it doesn't seem to have all those options
2. I don't have any v2s whatsoever so I'm flying blind based on what was written here to guess where it stores the settings and that it can load them. correctly

Looking at how it displays, using tool tips to display the meaning of items is a lot less optimal here. It seems like the optimal thing would be  `<select>` with a bonus option for to enter a value manually for any brave fool who wants to see if there's other possibilities.

**Comment by virmaior** [comment]
assuming it hasn't been manually edited can I read the current state from `/configs/.user_config` ? or is there some other way to get it / no way?

**Comment by adamfeldman** [comment]
For anyone else trying to enable the timestamp without using the Wyze app...I found the `[SETTING]` stanza in `configs/.parameters` and was able to edit it there and `reboot`.

**Reply by cpvolvo** [reply]
Maybe I am missing something, but I am looking in wz_mini/etc/configs/ and only have a .gitignore in there - am I looking in the wrong place?

**Reply by kohrar** [reply]
@cpvolvo, You should look under `/configs` on the Wyze Cam. It's a special directory on the running system, not the copy within the repo.

**Reply by cpvolvo** [reply]
Thank you! Makes sense
